### ISSUE TITLE
Shader resource links are probably hardcoded and throw error after changing config directory

### ISSUE THREAD

**Issue by Waldkoenig** [libretro/RetroArch#11022]
[INFO] Saved shader preset to D:\OneDrive\Games\ROM stuff\config\MAME\MAME.slangp.

[ERROR] [Vulkan]: Failed to load LUT "D:\OneDrive\Games\ROM stuff\config\MAME\:\shaders\shaders_slang\crt\shaders\crt-royale\TileableLinearApertureGrille15Wide8And5d5SpacingResizeTo64.png".

[ERROR] [Vulkan]: Failed to create preset: "D:\OneDrive\Games\ROM stuff\config\MAME\MAME.slangp".

[ERROR] [Vulkan]: Failed to create filter chain: "D:\OneDrive\Games\ROM stuff\config\MAME\MAME.slangp". Falling back to stock.

Other shaders without the lut part for example load and save fine.

**Comment by hizzlekizzle** [comment]
Did you move the shader preset? It indeed needs to have a connection to the LUT file, so if that location changes, it won't be able to find it.

**Comment by Waldkoenig** [comment]
No, I just moved the config-folder location and of course set the new one in RetroArch. 
But I read that a few versions ago the default save folder for created presets was changed from the shader folder to the config folder, so I guess that broke something.
I also downloaded updated all shader files with the online updater of course.

**Comment by Waldkoenig** [comment]
Are you saying I can't use the provided option if I want to use shaders?
When I say I moved the config folder I'm talking about the directory setting.

**Comment by Waldkoenig** [comment]
I just think that somewhere in your code the new path is constructed wrong, because this doesn't look proper:

D:\OneDrive\Games\ROM stuff\config\**MAME:\shaders**\shaders_slang\crt\shaders\crt-royale\TileableLinearApertureGrille15Wide8And5d5SpacingResizeTo64.png

**Comment by Waldkoenig** [comment]
Just for clarification. After loading a shader and playing around with the parameters everything works fine. Only when I save it as a preset it breaks.

**Comment by OctopusButtons** [comment]
> Yeah, if it's saving to the config directory and you moved that, all of the relative paths will be messed up.

I'm seeing an issue related to the colon.

1. I moved my shader directory, and config directory.  (Justified paranoia about Windows C:\ troubleshooting wipes/restores)
2. I correctly set the new locations in Settings > Directory.
3. I then found that my core preset shaders stopped working after that.
4.  I looked inside the `.slangp` files: (example) `#reference ":/shaders/CRT Newpixe e1.slangp"`
6. I had to manually change the `:` colons to `../..` (traditional relative folder structure path up)


That colon format breaks the path, until I manually changed it to `../..`  Is this a known thing, where the path format used for .slangp saves isn't compatible with changed directory locations?  If I understand it right, I thought that if the original format used `../../` instead of `:/` then it would have been fine even after moving the directory, as long as the relation is still "go up two folders from this one" (which it is in my case).

In other words, I kept the same folder structure, assuming my relative paths for shaders needed that, which is true, except the .slangp path was never coded with a relative path.  It was coded (problematically I think?) with a colon that resolves to *default* RetroArch directory.
